peer/brontide.go

```diff
@@ -3133,19 +3133,18 @@ func (p *Brontide) tryLinkShutdown(cid lnwire.ChannelID) error {
 		return ErrChannelNotFound
 	}
 
-	// Else, the link exists, so attempt to trigger shutdown. If this
-	// fails, we'll send an error message to the remote peer.
-	if err := chanLink.ShutdownIfChannelClean(); err != nil {
-		return err
-	}
-
-	// Next, we remove the link from the switch to shut down all of the
-	// link's goroutines and remove it from the switch's internal maps. We
-	// don't call WipeChannel as the channel must still be in the
-	// activeChannels map to process coop close messages.
-	p.cfg.Switch.RemoveLink(cid)
-
-	return nil
+	return chanLink.Flush(func() {
+		// Else, the link exists, so attempt to trigger shutdown. If
+		// this fails, we'll send an error message to the remote peer.
+		chanLink.ShutdownHtlcManager()
```

Comment:
I think we can hide this information from the caller, by executing this logic in the link if a new arg is set.

Reply:
Also `RemoveLink` will end up shutting down the main loop here: https://github.com/lightningnetwork/lnd/blob/5e369a02bb789140d286391edb7a86d78337dd25/htlcswitch/switch.go#L2580-L2581

So with that, I don't think we need `ShutdownHtlcManager` at all.